core/src/main/java/org/ehcache/core/EhcacheManager.java

```diff
@@ -117,6 +117,7 @@ public EhcacheManager(Configuration config, Collection<Service> services, boolea
     this.serviceLocator = new ServiceLocator(services.toArray(new Service[services.size()]));
     this.useLoaderInAtomics = useLoaderInAtomics;
     validateServicesConfigs();
+    resolveServices();
```

Comment:
So that the full set of services is loaded in to the ServiceLocator before maintenance is called.  Before this change it wasn't being done until init, which means maintenance won't see the whole set.